src/windows/Link.tsx

```diff
@@ -0,0 +1,149 @@
+/**
+* Link.tsx
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT license.
+*
+* RN Desktop-specific implementation of the cross-platform Link abstraction.
+*/
+
+import React = require('react');
+import RNW = require('react-native-windows');
+import {applyFocusableComponentMixin, FocusManagerFocusableComponent} from '../native-desktop/utils/FocusManager';
+import PropTypes = require('prop-types');
+
+import EventHelpers from '../native-common/utils/EventHelpers';
+import {Link as LinkCommon } from '../native-common/Link';
+
+const KEY_CODE_ENTER = 13;
+const KEY_CODE_SPACE = 32;
+
+const DOWN_KEYCODES = [KEY_CODE_SPACE, KEY_CODE_ENTER];
+const UP_KEYCODES = [KEY_CODE_SPACE];
+
+export interface LinkContext {
+    isRxParentAText?: boolean;
```

Comment:
See question above about isRxParentAText. I don't understand why this is needed in native.

Reply:
Basically it covers a case that *could* be covered by React Native for Windows (either by mitigating, or by implementing the right approach), but it currently doesn't.

Our RX.Link is using a plain RN.Text for iOs/Android, whereas for Windows we encapsulate one of those in a RNW.FocusableWindows control to give it keyboard focusing capability.
Problem is that currently inlining such a construct in a parent RN.Text (when embedding a link in a longer text) crashes RNW completely.

I chose to mitigate this for now by avoiding the RNW.FocusableWindows encapsulation when the parent is an RN.Text generating RX.Text, with the hope that a NYI XamlLink based view would come to help this scenario, or maybe all of them (there are some tradeoffs between accessibility and select-ability, so we're not sure of the solution yet).

But if this tracking is not allowed on RXP side, what I'm left with is removing RX.Link from the story for now, and we'll see what future reserves us.




---
In reply to: [159786422](https://github.com/Microsoft/reactxp/pull/404#discussion_r159786422) [](ancestors = 159786422)